<!-- Phidgets High Speed Encoder launch file -->

<launch>

  #### Nodelet manager ######################################################

  <node pkg="nodelet" type="nodelet" name="enc_manager"
    args="manager" output="screen" />

  #### High Speed Encoder Driver ############################################

  <node pkg="nodelet" type="nodelet" name="PhidgetsHighSpeedEncoderNodelet"
    args="load phidgets_high_speed_encoder/PhidgetsHighSpeedEncoderNodelet enc_manager"
    output="screen">

    # See README.md for more information on the parameters

    # optional param serial, default is -1
    <!-- <param name="serial" value="123456"/> -->

    # optional param hub_port, used if connected to a VINT hub
    <!-- <param name="hub_port" value="0"/> -->

    # optional param frame_id, default is "encode_link"
    <!-- <param name="frame_id" value="encoder_link"/> -->

    # optional param speed_filter_samples_len, default is 10
    <!-- <param name="speed_filter_samples_len" value="10" /> -->

    # optional param speed_filter_idle_iter_loops_before_reset, default is 1
    <!-- <param name="speed_filter_idle_iter_loops_before_reset" value="1" /> -->

    # optional param publish_rate, defaults to 0
    <!-- <param name="publish_rate" value="0" /> -->

    # optional param server_name, required to locate remote network servers
    <!-- <param name="server_name" value="myPhidgetServer" /> -->

    # optional param server_ip, required to locate remote network servers
    <!-- <param name="server_ip" value="0.0.0.0" /> -->
  </node>

</launch>
